libraries/core/src/main/kotlin/com/bumble/appyx/core/children/ChildNodeCreationManager.kt

```diff
@@ -49,7 +49,8 @@ internal class ChildNodeCreationManager<NavTarget : Any>(
                 when (keepMode) {
                     ChildEntry.KeepMode.KEEP -> {
                         navModelKeepKeys =
-                            (state.onScreen + state.offScreen).mapNotNullToSet { element -> element.key }
+                            state.onScreen.mapNotNullToSet { element -> element.key } +
```

Comment:
The Kotlin compiler reeeeeeally doesn't like `(state.onScreen + state.offScreen)`

I assume it has something to do with the generics, but even after attempting to change a lot of code to use `: Any` it still didn't work 🤔

Reply:
at least there was a workaround 🫠

Reply:
a new set.
also the definition of `navModelKeepKeys` is `val navModelKeepKeys: Set<NavKey<NavTarget>>` so it wouldn't compile unless it returned a set.